src/components/side-panel/common/types.ts

```diff
@@ -40,31 +36,31 @@ export type StyledDatasetTitleProps = {
 
 export type DatasetTitleProps = {
   dataset: KeplerTable;
-  showDatasetTable: Function;
   showDeleteDataset: boolean;
   onTitleClick?: () => void;
-  updateTableColor?: (d: string, c?: RGBColor) => void;
-  removeDataset?: (d: string) => void;
+  showDatasetTable?: typeof VisStateActions.showDatasetTable;
+  updateTableColor: typeof VisStateActions.updateTableColor;
+  removeDataset?: typeof openDeleteModal;
 };
 
 export type SourceDataCatalogProps = {
-  datasets: Datasets;
-  showDatasetTable: Function;
-  showDeleteDataset: boolean;
+  datasets: KeplerTable[];
```

Comment:
this should be `Datasets` an object, instead of an arrar

Reply:
fixed